File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left = 0
        right = len(height) - 1
        maxCapacity = 0
        while(left != right or height[left] != height[left]):
            currCapacity = min(height[left], height[right]) * (right - left)
            maxCapacity = max(maxCapacity, currCapacity)
            if(height[right] < height[left]):
                right -= 1
            else:
                left += 1
        return maxCapacity

        
        
        

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        if(height.empty()) return 0;
        int low = 0;
        int high = height.size() - 1;
        int maxArea = 0;
        while(high > low)
        {
            int area = min(height[high], height[low]) * (high - low);
            maxArea = max(maxArea, area);
            if (height[high] > height[low]) low++;
            else high--;
        }

        return maxArea;
    }
};